lua/plenary/async/util.lua

```diff
@@ -107,7 +107,8 @@ end
 function M.apcall(async_fn, ...)
   if a.is_leaf_function(async_fn) then
     local tx, rx = channel.oneshot()
-    local stat, ret = pcall(async_fn, vararg.rotate(tx, ...))
+    local nargs = select('#', ...) + 1
+    local stat, ret = pcall(async_fn, vararg.rotate(nargs, tx, ...))
```

Comment:
We can't rely on select here for the same reasons as `rotate`. Maybe we need to store the argc in the leaf table too so we can get it here, what do you think?

Reply:
Yeah, should be easy. Instead of a map of function:boolean, it can be function:number to serve an extra purpose with no extra overhead.

Reply:
that seems good